addons/mail/static/src/core/common/persona_model.js

```diff
@@ -89,6 +93,18 @@ export class Persona extends Record {
     isInternalUser = false;
     /** @type {luxon.DateTime} */
     write_date = Record.attr(undefined, { type: "datetime" });
+    group_id = Record.attr(undefined, {
+        compute() {
+            if (!this.group_id) {
+                return;
+            }
+            this.store.groupIdToPersonas.get(this.group_id)?.delete(this);
+            if (!this.store.groupIdToPersonas.has(this.group_id)) {
+                this.store.groupIdToPersonas.set(this.group_id, new Set());
+            }
+            this.store.groupIdToPersonas.get(this.group_id).add(this);
+        },
+    });
```

Comment:
`group_id` is replaced by `groups_id`, this is a `Record.many("ResGroups")` from comment above.

The `groupIdToPersonas` is replaced by a relational field on `ResGroups`: `ResGroups/personas`. This is defined as `inverse: "group_ids"`. By doing so, this will be automatically be filled by inserting data in `Persona/groups_id` thanks to formatted data you receive from server.

So you don't have to define compute method at all, it's just
```js
groups_id = Record.many("ResGroups", { inverse: ''personas" });
```